### PR TITLE
esp8266: Set reasonable date & time stamps for files.

### PR DESCRIPTION
**Pull request by robert-hh**
Changed files: esp8266/fatfs_port.c and extmod/vfs_fat.c
The time stamp is taken from the RTC for all newly generated
or changed files. RTC must be maintained separately.
The dummy time stamp set in vfs.stat() for the TLD is set to
Jan 1, 2000, avoiding invalid time values.

### CODE REVIEW

**Comment by robert-hh** [comment]
Defer a possible merge today; the seconds value is wrong - it has to be divided by 2. I'll fix this night.

File: esp8266/fatfs_port.c [hunk 1]
```diff
@@ -1,6 +1,41 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Josef Gajdusek
```
Comment (pfalcon) [inline esp8266/fatfs_port.c]: Please use @dpgeorge's copyright.

**Comment by pfalcon** [comment]
Please avoid introducing merged into pull requests, https://github.com/micropython/micropython/wiki/DevelWorkflow explains how.

File: esp8266/fatfs_port.c (hunk 1 above)
Comment (robert-hh) [inline esp8266/fatfs_port.c]: Oops. Copy & Paste omission. I'll fix it this evening

File: esp8266/fatfs_port.c [hunk 2]
```diff
@@ -1,6 +1,41 @@
...
+#include "timeutils.h"
+#include "modpybrtc.h"
 
 DWORD get_fattime(void) {
-    return 0;
+
+    uint64_t msecs = pyb_rtc_get_us_since_2000() / 1000;
+
+    timeutils_struct_time_t tm;
+    timeutils_seconds_since_2000_to_struct_time(msecs / 1000, &tm);
```
Comment (pfalcon) [inline esp8266/fatfs_port.c]: Another q: why do devide twice by 1000 instead of once by 1000000?

Comment (robert-hh) [inline esp8266/fatfs_port.c]: No reason. It was like that in the code I copied.

**Comment by robert-hh** [comment]
To much mess in the publishing. I'll restart

File: esp8266/fatfs_port.c (hunk 2 above)
Comment (pfalcon) [inline esp8266/fatfs_port.c]: I checked stmhal - it doesn't have such code, so what did you use?

Comment (robert-hh) [inline esp8266/fatfs_port.c]: Some of the RTC module code

**Pull request closed without merging**